OWML.ModHelper/Logging/ModSocketOutput.cs

```diff
@@ -1,59 +1,89 @@
+                ConnectToSocket();
             }
         }
 
-        public override void WriteLine(string s)
+        [Obsolete]
+        public override void WriteLine(params object[] objects)
         {
-            Logger?.Log(s);
-            CallWriteCallback(Manifest, s);
-            var message = $"{Manifest.Name}{NameMessageSeparator}{s}";
-            InternalWriteLine(message);
+            var line = string.Join(" ", objects.Select(o => o.ToString()).ToArray());
+            var type = MessageType.Message;
+            WriteLine(type, line, GetCallingMethodName(new StackTrace()));
         }
 
-        public override void WriteLine(params object[] objects)
+        public override void WriteLine(string line)
+        {
+            var type = MessageType.Message;
+            WriteLine(type, line, GetCallingMethodName(new StackTrace()));
+        }
+
+        public override void WriteLine(string line, MessageType type)
+        {
+            WriteLine(type, line, GetCallingMethodName(new StackTrace()));
+        }
+
+        private void WriteLine(MessageType type, string line, string senderType)
         {
-            WriteLine(string.Join(" ", objects.Select(o => o.ToString()).ToArray()));
+            Logger?.Log(line);
+            CallWriteCallback(Manifest, line);
+
+            var message = new SocketMessage
+            {
+                SenderName = Manifest.Name,
+                SenderType = senderType,
+                Type = type,
+                Message = line
+            };
+            var json = JsonConvert.SerializeObject(message);
+
+            WriteToSocket(json);
         }
 
-        private void CreateSocket()
+        private string GetCallingMethodName(StackTrace frame)
         {
-            var consolePortArgument = CommandLineArguments.GetArgument(Constants.ConsolePortArgument);
-            if (!int.TryParse(consolePortArgument, out var port))
+            try
             {
-                Logger?.Log("Error: Missing or incorrectly formatted console port argument");
-                return;
+                return frame.GetFrame(1).GetMethod().DeclaringType.Name;
             }
+            catch
+            {
+                return "";
+            }
+        }
 
+        private void ConnectToSocket()
+        {
             _socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
-            var ipAddress = IPAddress.Parse(LocalHost);
-            var endPoint = new IPEndPoint(ipAddress, port);
+            var ipAddress = IPAddress.Parse(Constants.LocalAddress);
+            var endPoint = new IPEndPoint(ipAddress, _port);
             _socket.Connect(endPoint);
         }
 
-        private void InternalWriteLine(string message)
+        private void WriteToSocket(string message)
         {
-            var bytes = Encoding.UTF8.GetBytes(message + Environment.NewLine);
+            var bytes = Encoding.UTF8.GetBytes(message);
```

Comment:
Can you keep the NewLine here? I think I remember you mentioning something about this, but I'm realizing now that this makes it a load easier to read the lines in the Mod Manager with less code changes. I tested it and it doesn't seem to affect the logs in the OWML console.